src/main/template/rds-instance.yaml

```diff
@@ -0,0 +1,243 @@
+    Properties:
+      GroupId: !Ref securityGroup
+      IpProtocol: tcp
+      FromPort: 3306
+      ToPort: 3306
+      SourceSecurityGroupId:
+        Fn::ImportValue:
+          !Sub "${vpcStackName}-bastion-security-group"
+
+  public5432Ingress:
+    Type: AWS::EC2::SecurityGroupIngress
+    Condition: createPublicPostgresIngress
+    Properties:
+      GroupId: !Ref securityGroup
+      IpProtocol: tcp
+      FromPort: 5432
+      ToPort: 5432
+      CidrIp: 0.0.0.0/0
+
+  private5432Ingress:
+    Type: AWS::EC2::SecurityGroupIngress
+    Condition: createPrivatePostgresIngress
+    Properties:
+      GroupId: !Ref securityGroup
+      IpProtocol: tcp
+      FromPort: 5432
+      ToPort: 5432
+      SourceSecurityGroupId:
+        Fn::ImportValue:
+          !Sub "${vpcStackName}-bastion-security-group"
+
+  subnetGroup:
+    Type: AWS::RDS::DBSubnetGroup
+    Properties:
+      DBSubnetGroupDescription: !Sub ${AWS::StackName}-database-subnet-group
+      SubnetIds:
+        - Fn::Select:
+          - 0
+          - Fn::Split:
+            - ","
+            - Fn::ImportValue:
+                Fn::If:
+                  - isPublic
+                  - !Sub "${vpcStackName}-public-subnets"
+                  - !Sub "${vpcStackName}-private-subnets"
+        - Fn::Select:
+          - 1
+          - Fn::Split:
+            - ","
+            - Fn::ImportValue:
+                Fn::If:
+                  - isPublic
+                  - !Sub "${vpcStackName}-public-subnets"
+                  - !Sub "${vpcStackName}-private-subnets"
+        - Fn::Select:
+          - 2
+          - Fn::Split:
+            - ","
+            - Fn::ImportValue:
+                Fn::If:
+                  - isPublic
+                  - !Sub "${vpcStackName}-public-subnets"
+                  - !Sub "${vpcStackName}-private-subnets"
+
+  dbInstance:
+    Type: AWS::RDS::DBInstance
+    Properties:
+      PubliclyAccessible:
+        Fn::If:
+          - isPublic
+          - true
+          - false
+      DBName: !Ref defaultSchemaName
+      DBInstanceIdentifier: !Sub ${AWS::StackName}-database
+      AllocatedStorage: !Ref storageSize
+      DBInstanceClass: !Ref instanceType
+      Engine: !Ref engine
+      MasterUsername: !Ref masterUsername
+      MasterUserPassword: !Ref masterUserPassword
+      DBSubnetGroupName: !Ref subnetGroup
+      VPCSecurityGroups:
+        - !Ref securityGroup
+
+Outputs:
+  JdbcUrl:
+    Description: The JDBC url to connect to the service database.
+    Value:
+      Fn::Join:
+        - ""
+        - - "jdbc:"
+          - !Ref engine
+          - "://"
+          - !GetAtt [ dbInstance, Endpoint.Address ]
+          - ":"
+          - !GetAtt [ dbInstance, Endpoint.Port ]
+          - "/"
+          - !Ref defaultSchemaName
+          - "?user="
+          - !Ref masterUsername
+          - "&password="
+          - !Ref masterUserPassword
+    Export:
+      Name: !Sub ${AWS::StackName}-jdbc-url
+
+  Host:
```

Comment:
Does host have a special purpose?

Seeing the exports for `-jdbc-url`, which is a convenience export for JDBC based clients, on the one side and  `-master-username` and `-master-user-password`, which simply hold "information about this stack", on the other side, I wonder if we should rather export
* address
* port
* master username
* master password
* engine
* default schema name

so dependants of this stack can use the information in any way they see fit rather than this stack making assumptions on how it's being used.

What do you think?

I admit, that would be a rather drastic shift. Up until now, our stacks have exposed convenient values that were optimized for the intended use.

Reply:
Why not both? We could have your list of values *and* the JDBC url.